### PR TITLE
Delete execution directory on flow completion.

### PR DESCRIPTION
**Pull request by reallocf**
Right now we store the project information, all user-generated intermediary files, and the job logs in the /executions/<execution_id> on the executor. The job logs have already been moved to the database and the project information will be re-copied for each subsequent execution, so storing this data on disk isn't necessary.

This change should hopefully free up some disk space. The only downside is we won't be able to get user's flow logs if they do not get uploaded to the database due to some error, but after discussing with @HappyRay we agreed that it's worth it to reclaim the disk space.

Tested on solo-server with both normal flows and nested flows.

### CODE REVIEW

**Comment by suvodeep-pyne** [comment]
@reallocf 
> The only downside is we won't be able to get user's flow logs if they do not get uploaded to the database due to some error, 
Is there a way to ensure that the delete only happens if we have a successful database copy?

**Comment by reallocf** [comment]
@suvodeep-pyne have any ideas on how to do this? I'm worried about adding complexity by having to pass around whether or not logs were uploaded successfully - particularly for nested flows.

@HappyRay what do you think? When we talked offline it seemed like this risk was worth it given how rarely database uploads fail, do you still think that's true?

**Comment by chengren311** [comment]
@reallocf thanks for doing this. What is the current way of cleaning execution dirs? As far as I could remember, they are cleaned up by cleaner thread inside flowrunnermannger. And it's a polling based mode which cleans up every 1 hour on directories not been modified for 1 day.  
Why not just making the upper limit(1 day) shorter?

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java [hunk 1]
```diff
@@ -221,6 +222,12 @@ public void run() {
       this.logger.info("Setting end time for flow " + this.execId + " to "
           + System.currentTimeMillis());
       closeLogger();
+      this.logger.info("Deleting execution directory " + this.execDir.toString());
```
Comment (HappyRay) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java]: Have you tested that this change works with embedded flows?

**Comment by HappyRay** [comment]
I am ok with delaying adding more complexity until there is evidence that it is a big problem. I haven't seen such evidence yet. If DB is down for an extended period of time, we have a bigger problem.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java (hunk 1 above)
Comment (reallocf) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java]: Yes! This still works with embedded flows :+1:

**Comment by reallocf** [comment]
After speaking offline with @suvodeep-pyne, we agreed it'd be better to set a configurable TTL for these files instead of deleting them. While we don't use these files often in our debugging (unless the problem is with uploading logs files to the DB) our users might use the /execution/<exec_id> directory to do their own testing then view the files via the HDFS viewer. While I don't think we have numbers regarding how often this is done, I can imagine users doing this.

I'll start working on the configurable TTL today, likely using the cleaner thread. @chengren311 I might bug you about how the thread works as I look more into it :smile:

**Comment by reallocf** [comment]
@HappyRay how does that sound to you?

**Comment by reallocf** [comment]
No longer going about this problem this way, so I'm closing this PR and opening another one with changes discussed offline.

**Pull request closed without merging**